core/src/main/java/org/mapfish/print/processor/jasper/JasperReportBuilder.java

```diff
@@ -77,28 +79,12 @@ File compileJasperReport(final File buildFile, final File jasperFile) throws JRE
         // another thread is reading it, use a temporary file as a target instead and
         // move it (atomic operation) when done. Worst case: we compile a file twice instead
         // of once.
-        File tmpBuildFile =
-            File.createTempFile(
-                "temp_", JASPER_REPORT_COMPILED_FILE_EXT, buildFile.getParentFile());
-
         LOGGER.info("Building Jasper report: {}", jasperFile.getAbsolutePath());
         LOGGER.debug("To: {}", buildFile.getAbsolutePath());
-        final Timer.Context compileJasperReport =
-            this.metricRegistry.timer(getClass().getName() + ".compile." + jasperFile).time();
-        try {
-          JasperCompileManager.compileReportToFile(
-              jasperFile.getAbsolutePath(), tmpBuildFile.getAbsolutePath());
-        } catch (JRValidationException e) {
-          LOGGER.error("The report '{}' isn't valid.", jasperFile.getAbsolutePath());
-          throw e;
-        } finally {
-          final long compileTime =
-              TimeUnit.MILLISECONDS.convert(compileJasperReport.stop(), TimeUnit.NANOSECONDS);
-          LOGGER.info("Report '{}' built in {}ms.", jasperFile.getAbsolutePath(), compileTime);
+        final String timerName = getClass().getName() + ".compile." + jasperFile;
+        try (Timer.Context compileJasperReport = this.metricRegistry.timer(timerName).time()) {
```

Comment:
For me, it looks that there is an error on how this timer is managed...

Reply:
We will check if the behavior of the timer change with the close.

Reply:
Some timers were never stopped. So as discussed, to make it consistent with the rest, I will close the timer rather than stop() and log.